python/src/nanoarrow/iterator.py

```diff
@@ -83,8 +84,42 @@ def itertuples(obj, schema=None) -> Iterable[Tuple]:
     return RowTupleIterator.get_iterator(obj, schema=schema)
 
 
+def iterrepr(obj, schema=None, max_width: int = 80) -> Iterable[str]:
```

Comment:
Hmm, how would you feel about approaching this a different way? Instead of 
```
    >>> for item in iterator.iterrepr(array, max_width=9):
    ...     print(item)
```
we could potentially do something like
```
    >>> for item in array:
    ...     print(item.to_string(max_width=9))
```
This would be more of a bottoms-up approach rather than top-down, where each data type has its own `to_string()` method that gets called and we don't need to keep track of the nested depth in an auxiliary class like `ReprIterator`